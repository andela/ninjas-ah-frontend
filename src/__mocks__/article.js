export default (Object.article = {
  title: 'my title well done',
  description: 'description of the article: well',
  body: 'body of the article: well'
});
