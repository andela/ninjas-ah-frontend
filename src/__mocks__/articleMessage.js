export default (Object.message = { message: 'article message' });
