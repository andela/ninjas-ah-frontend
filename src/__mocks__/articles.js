export default (Object.articles = [
  {
    author: { username: 'nivelo', bio: null, image: null },
    body: 'body',
    coverUrl: 'v1562366643/ninjas/ezzmjn0ak9b5gyzuzo6j.png',
    createdAt: '2019-07-05T22:43:57.384Z',
    description: 'ijihjhj',
    dislikes: null,
    favorited: false,
    favoritesCount: 0,
    id: 3,
    likes: null,
    rating: 0,
    readTime: '0',
    slug: 'how-to-conditionally-build-an-object-in-javascript-with-es6-m5mjxqor8sh',
    status: 'published',
    tagList: null,
    title: 'How to conditionally build an object in JavaScript with ES6',
    updatedAt: '2019-07-09T12:36:34.033Z',
    userId: 1
  }
]);
