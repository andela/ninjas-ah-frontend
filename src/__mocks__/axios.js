const mockAxios = jest.genMockFromModule('axios');

export const resolvedRequest = {
  data: {},
  status: 200,
  statusText: 'OK',
  headers: {},
  config: {}
};

export const rejectedRequest = {
  response: {
    data: { errors: { message: 'errors' } },
    status: 500,
    statusText: 'ERROR',
    headers: {},
    config: {}
  }
};

export default {
  get: jest.fn(() => Promise.resolve({ ...resolvedRequest })),
  post: jest.fn(() => Promise.resolve({ ...resolvedRequest, status: 201 })),
  put: jest.fn(() => Promise.resolve({ ...resolvedRequest })),
  delete: jest.fn(() => Promise.resolve({ ...resolvedRequest })),
  patch: jest.fn(() => Promise.resolve({ ...resolvedRequest })),
  create: jest.fn(() => mockAxios)
};
