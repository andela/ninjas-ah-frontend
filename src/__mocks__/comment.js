export default (Object.comment = { body: 'my title well done' });
