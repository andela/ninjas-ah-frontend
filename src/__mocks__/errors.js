export default (Object.errors = { message: 'can not updated article' });
