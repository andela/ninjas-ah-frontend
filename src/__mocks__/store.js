import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import initialState from '../store/initialState';
import apiMiddleware from '../middlewares/apiMiddleware';

export { initialState };
export const middlewares = [thunk, apiMiddleware];
export const mockStore = configureMockStore(middlewares);

export default mockStore(initialState);
