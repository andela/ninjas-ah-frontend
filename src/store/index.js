import 'dotenv/config';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import thunk from 'redux-thunk';
import initialState from './initialState';
import reducers from '../reducers';
import apiMiddleware from '../middlewares/apiMiddleware';

const middlewares = [thunk, apiMiddleware];

export default createStore(
  combineReducers(reducers),
  initialState,
  composeWithDevTools(applyMiddleware(...middlewares))
);
