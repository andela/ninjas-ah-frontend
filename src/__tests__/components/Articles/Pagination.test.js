import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { mockStore, initialState } from '../../../__mocks__/store';
import {
  Pagination as PaginationComponent,
  mapStateToProps
} from '../../../components/Articles/Pagination/Pagination';
import { shallow, mount } from '../../../../config/enzymeConfig';

describe('<Pagination />', () => {
  const state = {
    buttons: ['button'],
    pageNumber: 1,
    getAllArticles: jest.fn(),
    displayButtons: jest.fn(),
    paginateArticles: jest.fn()
  };
  const component = shallow(<PaginationComponent {...state} />);
  it('should create <PaginationComponent /> snapshot ', () => {
    expect(component).toMatchSnapshot();
  });
  it('should trigger submit button ', () => {
    component.instance().displayButtons(40);
  });
  it('should trigger close message ', () => {
    component.setProps({ limit: 1, offset: 10, label: 1 });
    component
      .find('.pagination button')
      .at(0)
      .simulate('click');
    component.instance().paginateArticles({ limit: 1, offset: 10, label: 1 });
  });
});
