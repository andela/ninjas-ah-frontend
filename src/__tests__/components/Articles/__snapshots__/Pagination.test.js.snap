// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should create <PaginationComponent /> snapshot  1`] = `ShallowWrapper {}`;
